src/Octoshift/Commands/WaitForMigrationCommandBase.cs

```diff
@@ -44,7 +47,63 @@ public async Task Handle(string migrationId, string githubPat = null, bool verbo
     {
         _log.Verbose = verbose;
 
+        if (migrationId is null)
+        {
+            throw new ArgumentNullException(nameof(migrationId));
+        }
+
+        if (!migrationId.StartsWith(REPO_MIGRATION_ID_PREFIX) && !migrationId.StartsWith(ORG_MIGRATION_ID_PREFIX))
+        {
+            throw new OctoshiftCliException($"Invalid migration id: {migrationId}");
+        }
+
         var githubApi = _githubApiFactory.Create(targetPersonalAccessToken: githubPat);
+
+        if (migrationId.StartsWith(REPO_MIGRATION_ID_PREFIX))
+        {
+            await WaitForRepositoryMigration(migrationId, githubPat, githubApi);
+        }
+        else
+        {
+            await WaitForOrgMigration(migrationId, githubPat, githubApi);
+        }
+    }
+
+    private async Task WaitForOrgMigration(string migrationId, string githubPat, GithubApi githubApi)
+    {
+        var state = await githubApi.GetOrganizationMigrationState(migrationId);
+
+        _log.LogInformation($"Waiting for org migration (ID: {migrationId}) to finish...");
+
+        if (githubPat is not null)
+        {
+            _log.LogInformation($"{GithubPat.GetLogFriendlyName()}: ***");
+        }
+
+        while (true)
+        {
+            if (OrganizationMigrationStatus.IsSucceeded(state))
+            {
+                _log.LogSuccess($"Migration {migrationId} succeeded");
+                return;
+            }
+
+            if (OrganizationMigrationStatus.IsFailed(state))
+            {
+                _log.LogError($"Migration {migrationId} failed");
```

Comment:
```suggestion
               throw new OctoshiftCliException($"Migration {migrationId} failed");
```

If we throw an exception the error will automatically get logged, but the command will also exit with a non-zero exit code. It's also consistent with what we do when waiting for repo migrations.